server/tserver/src/main/java/org/apache/accumulo/tserver/log/TabletServerLogger.java

```diff
@@ -308,11 +300,44 @@ public void run() {
                 log.warn("Failed to delete a WAL that failed to open", e);
               }
             }
+            
+            continue;
+          }
+          
+          String fileName = alog.getFileName();
+          log.debug("Created next WAL " + fileName);
+          
+          try {
+            tserver.addNewLogMarker(alog);
+          } catch (Exception t) {
+            log.error("Failed to add new WAL marker for " + fileName, t);
+              
+            try {
+              alog.close();
```

Comment:
It would be really nice to add a comment here that the file is intentionally not being deleted and why that is. Could also put the issue number in the comment.